src/main/java/org/junitpioneer/jupiter/RetryingTestExtension.java

```diff
@@ -147,6 +158,20 @@ private boolean expectedException(Throwable exception) {
 			return Arrays.stream(expectedExceptions).anyMatch(type -> type.isInstance(exception));
 		}
 
+		private void suspendFor(int millis) {
+			if (millis < 1) {
+				return;
+			}
+
+			try {
+				TimeUnit.MILLISECONDS.sleep(millis);
+			}
+			catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				throw new IllegalStateException("Cannot suspend execution after retry", e);
```

Comment:
```suggestion
				throw new IllegalStateException("Thread interrupted during retry suspension.", e);
```

Bonus points for renaming `e` to `ex`. (Old habit from writing code that deals with events _and_ exceptions. 😉 )